includes/User.php

```diff
@@ -5007,6 +5007,8 @@ protected function saveOptions() {
 
 		$dbw->upsert('user_properties', $insertRows, [], self::$PROPERTY_UPSERT_SET_BLOCK);
 
+		$this->userPreferences()->save($this->getId());
```

Comment:
Yeah, I think we should do that to properly separate preferences from options. Maybe it should be a function `savePreferences` that saves the preferences in the db and also invokes the hook?